test/features/environment.py

```diff
@@ -0,0 +1,94 @@
+                        Should be used with "-y" option to avoid SSH key be
+                        overwritten
+
+  Network configuration:
+    Options for configuring the network and messaging layer.
+
+    -i IF, --interface=IF
+                        Bind to IP address on interface IF
+    -u, --unicast       Configure corosync to communicate over unicast (UDP),
+                        and not multicast. Default is multicast unless an
+                        environment where multicast cannot be used is
+                        detected.
+    -A IP, --admin-ip=IP
+                        Configure IP address as an administration virtual IP
+    -M, --multi-heartbeats
+                        Configure corosync with second heartbeat line
+    -I, --ipv6          Configure corosync use IPv6
+    --qdevice=QDEVICE   QDevice IP
+    --qdevice-port=QDEVICE_PORT
+                        QDevice port
+    --qdevice-algo=QDEVICE_ALGO
+                        QDevice algorithm
+    --qdevice-tie-breaker=QDEVICE_TIE_BREAKER
+                        QDevice algorithm
+
+  Storage configuration:
+    Options for configuring shared storage.
+
+    -p DEVICE, --partition-device=DEVICE
+                        Partition this shared storage device (only used in
+                        "storage" stage)
+    -s DEVICE, --sbd-device=DEVICE
+                        Block device to use for SBD fencing
+    -o DEVICE, --ocfs2-device=DEVICE
+                        Block device to use for OCFS2 (only used in "vgfs"
+                        stage)
+
+
+Stage can be one of:
+    ssh         Create SSH keys for passwordless SSH between cluster nodes
+    csync2      Configure csync2
+    corosync    Configure corosync
+    storage     Partition shared storage (ocfs2 template only)
+    sbd         Configure SBD (requires -s <dev>)
+    cluster     Bring the cluster online
+    vgfs        Create volume group and filesystem (ocfs2 template only,
+                requires -o <dev>)
+    admin       Create administration virtual IP (optional)
+
+Note:
+  - If stage is not specified, the script will run through each stage
+    in sequence, with prompts for required information.
+  - If using the ocfs2 template, the storage stage will partition a block
+    device into two pieces, one for SBD, the remainder for OCFS2.  This is
+    good for testing and demonstration, but not ideal for production.
+    To use storage you have already configured, pass -s and -o to specify
+    the block devices for SBD and OCFS2, and the automatic partitioning
+    will be skipped.'''
+
+def before_feature(context, feature):
+    if feature.name == "HA bootstrap process":
+        context.init_help_message = CLUSTER_INIT_HELP_MESSAGE
+
+def before_scenario(context, scenario):
+    if utils.check_service_active('corosync.service', 'local'):
+        utils.run_command('crm cluster stop')
+    if scenario.name == "No overwrite ssh key but append to authorized_keys":
```

Comment:
This looks quite risky. I wouldn't like to have specific conditions for particular test scenarios. Maybe we can perform this action in the test itself